### PR TITLE
Run unit tests on github action

### PR DESCRIPTION
**Pull request by deguif**
See result here: https://github.com/deguif/Elastica/actions/runs/395847967 or at the end of the PR in the checks section.

I had to make several changes in tests to fix unit tests that were functional tests, here's the result without the changes on test classes: https://github.com/deguif/Elastica/runs/1480767481

### CODE REVIEW

File: .github/workflows/continuous-integration.yaml [hunk 1]
```diff
@@ -18,3 +18,53 @@ jobs:
...
+          key: '${{ runner.os }}-composer-php${{ matrix.php }}-${{ hashFiles(''**/composer.json'') }}'
+          restore-keys: |
+            ${{ runner.os }}-composer-php${{ matrix.php }}-
+            ${{ runner.os }}-composer-
+
+      - name: 'Update dependencies'
+        run: 'composer update --prefer-dist --no-interaction --no-progress --ansi ${{ matrix.composer_flags }}'
+
+      - name: 'Run tests'
+        run: 'vendor/bin/phpunit --group unit'
```
Comment (deguif) [inline .github/workflows/continuous-integration.yaml:70]: Just running unit test for the moment as booting a functional stack is more complicated. Will be done in a following PR if you don't mind.

File: tests/Query/TermsTest.php [hunk 2]
```diff
@@ -9,10 +9,12 @@
 
 /**
  * @internal
- * @group unit
```
Comment (deguif) [inline tests/Query/TermsTest.php:12]: Adding this annotation at the class level, mark all methods as unit test.

**Comment by deguif** [comment]
It's blazing fast (without the functional tests part).
Screenshot 2020-12-02 at 12 02 51

File: tests/IndexTest.php [hunk 3]
```diff
@@ -626,17 +688,19 @@ public function testCreateWithInvalidOption(): void
...
-        $index = $client->getIndex($indexName);
+        $client = $this->createMock(Client::class);
+        $index = new Index($client, 'test');
 
         $opts = ['testing_invalid_option' => true];
         $index->create([], $opts);
     }
 
+    /**
+     * @group unit
```
Comment (ruflin) [inline tests/IndexTest.php:699]: unit?

File: tests/PipelineTest.php [hunk 4]
```diff
@@ -24,7 +25,8 @@ public function testProcessor(): void
         $rename = new Rename('foo', 'target.field');
         $set = new Set('field4', 324);
 
-        $processors = new Pipeline($this->_getClient());
+        $client = $this->createMock(Client::class);
```
Comment (ruflin) [inline tests/PipelineTest.php:28]: Can you share some background on this change?

File: tests/Query/ParentIdTest.php [hunk 5]
```diff
@@ -124,7 +124,7 @@ public function testParentId(): void
     }
 
     /**
-     * @group unit
+     * @group functional
```
Comment (ruflin) [inline tests/Query/ParentIdTest.php:127]: 🤦 Nice catch.

**Review by ruflin — APPROVED** [review]
Awesome. Left one question but should not block merging this PR! Please get it in.

File: tests/IndexTest.php (hunk 3 above)
Comment (deguif) [reply tests/IndexTest.php:699]: Yes unit as there is no call done on the elasticsearch stack. Mocking a client is sufficient as it's not used for this test

File: tests/PipelineTest.php (hunk 4 above)
Comment (deguif) [reply tests/PipelineTest.php:28]: Yes a client mock is sufficient here to satisfy the pipeline object constructor.
No call is made on the elasticsearch server, as only the `toArray()` method is tested here.

**Comment by deguif** [comment]
I just added a changelog.

**Pull request merged**